tests/robot-tests/scripts/get_auth_tokens.py

```diff
@@ -37,74 +37,109 @@ def get_identity_info(url, email, password, first_name="Bau1", last_name="EESADM
     chrome_options.add_argument('--log-level=\3')
     driver = webdriver.Chrome(options=chrome_options)
 
-    try:
-        driver.get(url)
-
-        wait_until_page_contains_xpath(driver, '//button[contains(text(), "Sign in")]')
-        driver.find_element_by_xpath('//button[contains(text(), "Sign in")]').click()
-
+    if identity_provider == 'KEYCLOAK':
         try:
```

Comment:
You could pull some of this code into functions to make the provider logic possible to read at a glance. Something like:
```
if id_provider == KEYCLOAK:
  login_with_keycloak(driver, url, email, password, whatever else)
elseif id_provider == AZURE:
  login_with_azure(...)
else:
  throw exception
```